libs/testdiff/replacement.go

```diff
@@ -27,11 +28,74 @@ var (
 	devVersionRegex = regexp.MustCompile(`0\.0\.0-dev(\+[a-f0-9]{10,16})?`)
 )
 
-type Replacement struct {
+type Replacement interface {
+	// Perform the replacement on the input string and return the result.
+	Replace(s string) string
+
+	// Returns debug string that enumerates the replacement.
+	Debug() string
+}
```

Comment:
I prefer having single plain old struct here, easy to reason about, load form config, serialize etc.

We can keep using Replacement as a basic and easy to understand building block and implement UUID-i functionality one layer above.